guides/keras_cv/writing_custom_image_augmentations_with_baseimageaugmentationlayer.py

```diff
@@ -0,0 +1,435 @@
+"""
+Title: Custom Image Augmentations with BaseImageAugmentationLayer
+Author: [lukewood](https://lukewood.xyz)
+Date created: 2022/04/26
+Last modified: 2022/04/26
+Description: Use BaseImageAugmentationLayer to implement custom data augmentations.
+"""
+
+"""
+## Overview
+Data augmentation is an integral part of training any robust computer vision model.
+While KerasCV offers a plethora of prebuild high quality data augmentation techniques,
+you may still want to implement your own custom technique.
+KerasCV offers a helpful base class for writing data augmentation layers:
+`BaseImageAugmentationLayer`.
+Any augmentation layer built with `BaseImageAugmentationLayer` will automatically be
+compatible with the KerasCV `RandomAugmentationPipeline` class.
```

Comment:
Consider hyperlinking `RandomAugmentationPipeline` and some of the notable augmentation layer classes we have for easier navigation.